### ISSUE TITLE
Improve db migration from version 3 to 4 when an old fork (e.g. KoboCollect) is also installed

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#997]
#### Problem description
There's a crash that can happen when migrating from db versions 3 to 4 because it tries to create columns that are already there. This can happen when there's a Collect-derived tool like Kobo on the device (see [this thread](https://groups.google.com/forum/#!topic/opendatakit/vOSPGNWQJGc)) or when going back to a version before the db migration and trying to upgrade again. 

#### Steps to reproduce the problem
1. Install and run a version of Collect > 1.4.15 
1. Go back to a version of Collect < 1.4.15 (e.g. `git checkout v1.4.13`)
1. Install and run a version of Collect > 1.4.15

#### Expected behavior
The migration should skip columns that already exist and not crash when trying to upgrade again.

#### Other information 
v1.4.15 added a `deletedOn` column to support View Sent Forms.

```
Exception android.database.sqlite.SQLiteException: Can't upgrade read-only database from version 3 to 4: /storage/emulated/0/odk/metadata/instances.db
org.odk.collect.android.database.ODKSQLiteOpenHelper.getReadableDatabase (ODKSQLiteOpenHelper.java:212)
org.odk.collect.android.provider.InstanceProvider.query (InstanceProvider.java:162)
android.content.ContentProvider.query (ContentProvider.java:980)
android.content.ContentProvider$Transport.query (ContentProvider.java:213)
android.content.ContentResolver.query (ContentResolver.java:484)
android.content.ContentResolver.query (ContentResolver.java:428)
org.odk.collect.android.dao.InstancesDao.getInstancesCursor (InstancesDao.java:228)
org.odk.collect.android.dao.InstancesDao.getSavedInstancesCursor (InstancesDao.java:106)
org.odk.collect.android.dao.InstancesDao.getSavedInstancesCursor (InstancesDao.java:112)
org.odk.collect.android.fragments.DataManagerList.getCursor (DataManagerList.java:154)
org.odk.collect.android.fragments.DataManagerList.setupAdapter (DataManagerList.java:137)
org.odk.collect.android.fragments.DataManagerList.onViewCreated (DataManagerList.java:77)
```

**Comment by yanokwa** [comment]
#1000 fixes the case when you are only using ODK Collect. I expect that if you have KoboCollect, you might still get crashes due to it acquiring a lock on the DB. We'll keep this issue open for now to watch for those kinds of crashes.

**Comment by lognaturel** [comment]
Closing this in favor of #1278 since the issue described here is fixed and #1278 shows an example of the class of problem @yanokwa mentioned.
